File: src/features/typing/data/go/set_matrix_zeroes.go
package main

func setZeroes(matrix [][]int) {
	rows := make(map[int]bool)
	cols := make(map[int]bool)
	for i, row := range matrix {
		for j, val := range row {
			if val == 0 {
				rows[i] = true
				cols[j] = true
			}
		}
	}
	for i, row := range matrix {
		for j := range row {
			if rows[i] || cols[j] {
				matrix[i][j] = 0
			}
		}
	}
}

File: src/features/typing/data/dart/set_matrix_zeroes.dart
class Solution {
  void setZeroes(List<List<int>> matrix) {
    final rows = <int>{};
    final cols = <int>{};
    for (int i = 0; i < matrix.length; i++) {
      for (int j = 0; j < matrix[i].length; j++) {
        if (matrix[i][j] == 0) {
          rows.add(i);
          cols.add(j);
        }
      }
    }
    for (int i = 0; i < matrix.length; i++) {
      for (int j = 0; j < matrix[i].length; j++) {
        if (rows.contains(i) || cols.contains(j)) {
          matrix[i][j] = 0;
        }
      }
    }
  }
}
